### DISCUSSION TITLE
password reset request

### DISCUSSION THREAD

**Discussion by breakingbad45** [pocketbase/pocketbase#5876] [Q&A]
How to implement this ?that user send a password reset request  send a email to registered email user confirm than change password?

**Comment by chnfyi** [comment]
It's already implemented, you need to custom your collection password reset email template in the dashboard (_Edit collection > Options_), then :

```javascript
// Send a password reset request to the user email (if registered)
// In this email the user clicks on the link that redirects on your password change page
await pb.collection('users').requestPasswordReset('test@example.com');

// On your password change page use the token you extract from the url
// and both new password and new password confirmation
await pb.collection('users').confirmPasswordReset('TOKEN', '1234567890', '1234567890');
``` 

If the above doesn't help, please try to clearly elaborate on what you are trying to do.

**Comment by breakingbad45** [comment]
```
  const handleSubmit = async (e) => {
    e.preventDefault();

    try {
      const response = await pb.collection("users").requestPasswordReset(email);
      console.log(response); // Debugging output
    //   setStatusMessage("Password reset email sent successfully! Please check your inbox.");
    //   setStatusType("success");
    } catch (error) {
      console.error(error); // Debugging output
    //   setStatusMessage("Failed to send password reset email. Please try again.");
    //   setStatusType("error");
    }
  };
```

if the email not found still its response come true.but in admin ui log there show the error.how to fix it?

**Reply by ganigeorgiev** [reply]
Success 204 response on nonexisting user is intentional as a very rudimentary measure to prevent emails enumeration (this apply also for the request verification endpoint) and I think it is a standard practice used by many other services too, especially when there is no captcha.

I recommend changing your success message to include the submitted email address in case of a typo so users can notice it and send a new request if needed, e.g. "Password reset email was sent to EMAIL. Please check your inbox".
